libraries/aws/shadow/src/aws_iot_shadow_operation.c

```diff
@@ -184,8 +184,80 @@ IotMutex_t _AwsIotShadowPendingOperationsMutex;
 
 /*-----------------------------------------------------------*/
 
-static bool _shadowOperationMatch( const IotLink_t * pOperationLink,
-                                   void * pMatch )
+_shadowOperationType_t _AwsIotShadow_IntToShadowOperationType( int32_t n )
```

Comment:
Why take a signed parameter here?  If you used `uint32_t n` here, then line 214 would just be:

```c
AwsIotShadow_Assert( n < 5 );
```

Same comment applies on `_AwsIotShadow_IntToShadowCallbackType()`